### DISCUSSION TITLE
Retry calls using the Retry-After header.

### DISCUSSION THREAD

**Discussion by felipecauefraga** [TanStack/query#6565] [General] [answered]
Hi! I'm testing React Query to enhance the performance of my app. I would like to utilize the Retry-After header (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) to obtain its value and use this time delay for retrying calls. Is there a way to configure React Query to incorporate this header for retry functionality?

**Comment by TkDodo** [comment, marked as answer]
React Query knows nothing about http headers because it's not tied to http at all. It's a promise based state manager, which means you can do anything as long as your `queryFn` returns a resolved or rejected promise.

In your case, here's what I would do:

- in the `querFn`, make your api call with your favourite tool.
- catch errors inside the `queryFn`
- look at the response headers and read the `Retry-After` header
- re-throw a custom error that has a property `retryAfter` you can look at
- implement the `retryDelay` function to look at the custom error and return a number in millis for how long to wait, something like:

```
useQuery({
  queryKey,
  queryFn,
  retryDelay: (failureCount, error) => error.retryAfter * 1000
})
```

**Reply by felipecauefraga** [reply]
Hey @TkDodo, thanks! I'll give this solution a shot.
